src/test/java/ro/secur/auth/security/AuthenticationFilterTest.java

```diff
@@ -0,0 +1,196 @@
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@SpringBootTest
+@AutoConfigureMockMvc
+public class AuthenticationFilterTest {
+
+    public static final String SECRET_KEY = "madkagdfgkjfhjrkojiurgoijdsgjkldfgdislfjspfo[fasdkfodkgdosmgbakw[pfkb";
+
+    @Mock
+    private AuthenticationManager authManagerMock;
+
+    @Mock
+    private JwtConfiguration jwtConfigMock;
+
+    @Mock
+    private MockHttpServletRequest requestMock;
+
+    @Mock
+    private MockHttpServletResponse responseMock;
+
+    private MockMvc mockMvc;
+
+    private AuthenticationFilter authFilter;
+
+    @Autowired
+    private WebApplicationContext webApplicationContext;
+
+    @BeforeEach
+    public void init() throws JSONException {
+
+        mockJwtConfig();
+        authFilter = new AuthenticationFilter(authManagerMock, jwtConfigMock);
+
+        mockMvc = MockMvcBuilders.webAppContextSetup(webApplicationContext)
+                .addFilter(authFilter)
+                .build();
+    }
+
+    private void mockJwtConfig() {
+        when(jwtConfigMock.secretKey()).thenReturn(Keys.hmacShaKeyFor(SECRET_KEY.getBytes()));
+        when(jwtConfigMock.getTokenExpirationDays()).thenReturn(14);
+        when(jwtConfigMock.getAuthorizationHeader()).thenReturn(HttpHeaders.AUTHORIZATION);
+        when(jwtConfigMock.getTokenPrefix()).thenReturn("Bearer ");
+    }
+
+    private void mockRequest(JSONObject jsonRequestBody) {
+        requestMock = new MockHttpServletRequest();
+        requestMock.setContentType(MediaType.APPLICATION_JSON);
+        requestMock.setCharacterEncoding("UTF-8");
+        requestMock.setContent(jsonRequestBody.toString().getBytes());
+    }
+
+    private JSONObject createJsonObject(String username, String password) throws JSONException {
+        JSONObject jsonRequestBody = new JSONObject();
+
+        jsonRequestBody.put("username", username);
+        jsonRequestBody.put("password", password);
+
+        return jsonRequestBody;
+    }
+
+    private void mockAuthManagerReturnsAuth(Authentication expectedAuth) {
+
+        when(authManagerMock.authenticate(any(Authentication.class))).thenReturn(expectedAuth);
+    }
+
+    private void mockAuthManagerReturnsException() {
+
+        when(authManagerMock.authenticate(any(Authentication.class))).thenThrow(AuthenticationException.class);
+    }
+
+    private void mockAuthManagerReturnsBadCredentials() {
+
+        when(authManagerMock.authenticate(any(Authentication.class))).thenThrow(BadCredentialsException.class);
+    }
+
+    @SneakyThrows
+    @Test
+    public void whenAttemptAuthentication_returnAuth() {
+
+        Authentication expectedAuth = new UsernamePasswordAuthenticationToken("test", "pass",
+                Collections.singletonList(new SimpleGrantedAuthority(Role.USER.toString())));
```

Comment:
please use name() instead of toString method. The reason behind this is the fact that toString method can me overriden. Change that into all test classes.